generated/nifake/nifake_library.cpp

```diff
@@ -378,11 +378,7 @@ ViStatus NiFakeLibrary::GetError(ViSession vi, ViStatus* errorCode, ViInt32 buff
   if (!function_pointers_.GetError) {
     throw nidevice_grpc::LibraryLoadException("Could not find niFake_GetError.");
   }
-#if defined(_MSC_VER)
```

Comment:
Note:
There are some functions in nifake metadata for which codegen_method are private. But those functions were also present in nifake.h. But in the case of DCPower, the `ParseChannelCount` function was not present in nidcpower.h. So I had modified mako script to use only function pointers for private functions. 

But if we don't want to apply that rule for all private functions, we have following alternative approaches:
1. Introduce a new property called 'use_function_pointer' to the metadata for ParseChannelCount API. Only if that property is set to True, then use only function pointer.
2. Second alternative is introduce a new value for codegen_method something like - `'codegen_method':'internal'`. And only for this internal codegen_methods, use only function pointers.

But for me, the current approach of using only function_pointers for all private functions looked more cleaner. But if reviewers ask to use any other approach, I can modify my code.

Reply:
@doshirohan , @ganeshnie , Is it okay to use only function_pointers for these private functions of nifake? Any thoughts?

Note: The main purpose of using static linking at compile-time was to ensure the syntax of generated code is correct and catch any errors in syntax earlier during compile time itself. But for private functions, would it be okay to not get that advantage?

Reply:
Irrespective of whether function is private or public, static linking would help in identifying syntax issues early in development (during compilation) than later at runtime when someone tries to use a public API which exercises the API in question. I like simplicity of this approach and given number of private APIs we have is near 0, I think this approach should be ok. Later if we need to use many such "private" APIs (which are present in header but we don't want to expose to user), we can introduce new tag for that then.

Reply:
I too think we should try to keep the capability of catching the compile errors.  I kind of like both the approaches Kruthi mentioned. Having something of an attribute saying `call_by_function_pointer_only` is what I would recommend.

Reply:
For gRPC, since we want to expose all the APIs which are present in C documentation, the chances of encountering APIs which are present in C doc but which need to be made private is almost nill for any driver. For NIFake, I don't know why those APIs were kept as private even if they were present in C header file. May be we can get opinions from @jonlugoatx and @reckenro on this and agree on a common approach.

Reply:
In nimi-python, they marked many functions as private because they wanted to expose "fancy" versions that were more native to python. We removed those fancy functions and made the private ones public. So I think it's fair to say that we have a different definition of private. For `grpc-device`, a private function should only be used for ones not in public headers.

Reply:
And I think for ni-fake specifically, some methods were left as 'private' because we didn't get to them or didn't deem them different enough from other ni-fake methods we were unit testing to fully generate. I don't think we did a pass over the functions in the metadata to equate "public in C header" => "codegen_method: public" for the function metadata for ni-fake like we did with the other driver metadata. We could update ni-fake's metadata to make more (if not all) the private methods public but we just didn't get around to it.